dotnet/src/SemanticKernel.Skills/Skills.Grpc/GrpcOperationRunner.cs

```diff
@@ -0,0 +1,253 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Net.Client;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Skills.Grpc.Model;
+using ProtoBuf;
+using Skills.Grpc;
+
+namespace Microsoft.SemanticKernel.Skills.Grpc;
+
+/// <summary>
+/// Runs gRPC operation runner.
+/// </summary>
+internal class GrpcOperationRunner : IGrpcOperationRunner
+{
+    /// <summary>
+    /// Name of 'address' argument used as override for the address provided by gRPC operation.
+    /// </summary>
+    internal const string AddressArgumentName = "address";
+
+    /// <summary>
+    /// An instance of the HttpClient class.
+    /// </summary>
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Creates an instance of a <see cref="GrpcOperationRunner"/> class.
+    /// </summary>
+    /// <param name="httpClient">An instance of the HttpClient class.</param>
+    public GrpcOperationRunner(HttpClient httpClient)
+    {
+        this._httpClient = httpClient;
+    }
+
+    /// <inheritdoc/>
+    public async Task<object> RunAsync(GrpcOperation operation, IDictionary<string, string> arguments, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(operation, $"The {nameof(operation)} parameter is not set to an instance of an object.");
+        Verify.NotNull(arguments, $"The {nameof(arguments)} parameter is not set to an instance of an object.");
```

Comment:
The goal of the Verify class is providing contextual and actionable error messages.
If you want to throw generic/default error messages pls use ArgumentNullException.ThrowIfNull, although that will make debugging harder.

Reply:
Got it, make sense, I did not look at this aspect and considered them as contract constraints validation mechanism.